### PR TITLE
Add JAX + TPU Commons path

### PR DESCRIPTION
**Pull request by jrplatin**
## Description
In this PR, I do the following
* Add the ability to run the JAX path on TPU Commons
* Add an hourly test for that path, which runs both Llama3.1-8B and Qwen2.5-1.5B
* Modify the `docker/DockerfileTPUCommon.tpu`, `scripts/scheduler/build_tpu_commons_image.sh` , and `scripts/scheduler/create_job.sh` to accept new `TPU_COMMONS_TPU_BACKEND_TYPE` and `VLLM_TORCHAX_ENABLED` arguments

Updated the CodeHash to include the TPU Backend Type image

## Testing
Verified JAX + TPU Commons ([server](https://storage.mtls.cloud.google.com/vllm-cb-storage2/job_logs/04279223-a095-491c-92ec-9207384814ff/static_vllm_log.txt) log and client [log](https://storage.mtls.cloud.google.com/vllm-cb-storage2/job_logs/04279223-a095-491c-92ec-9207384814ff/static_bm_log.txt)) and TPU Commons + TorchAX ([sever](https://storage.mtls.cloud.google.com/vllm-cb-storage2/job_logs/9a2c6550-ce3d-4e95-a8ea-7ef9406f8b54/static_vllm_log.txt) log and [client](https://storage.mtls.cloud.google.com/vllm-cb-storage2/job_logs/9a2c6550-ce3d-4e95-a8ea-7ef9406f8b54/static_bm_log.txt) log)

### CODE REVIEW

File: scripts/scheduler/create_job.sh [hunk 1]
```diff
@@ -12,6 +12,7 @@ CODE_HASH="${2:-}"  # optional
 JOB_REFERENCE="${3:-}"
 RUN_TYPE="${4:-"MANUAL"}"
 REPO="${5:-"DEFAULT"}"
+TPU_COMMONS_TPU_BACKEND_TYPE="${6:-"torchax"}"
```
Comment (QiliangCui) [inline scripts/scheduler/create_job.sh:15]: Do you want to add an hourly run directly in https://github.com/QiliangCui/bm-infra/blob/main/scripts/scheduler/hourly_run.sh? 

take this as example

```
echo "./scripts/scheduler/create_job.sh ./cases/hourly_torchax.csv \"\" $TAG HOURLY_TORCHAX TPU_COMMONS"
./scripts/scheduler/create_job.sh ./cases/hourly_torchax.csv "" $TAG HOURLY_TORCHAX TPU_COMMONS
```

Or, it is OK to do this in next change. :) Just don't foget it.

Comment (jrplatin) [reply scripts/scheduler/create_job.sh:15]: Will add! Just wanted to make sure everything else looked good first

File: cases/hourly_jax.csv [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Device,Model,MaxNumSeqs,MaxNumBatchedTokens,TensorParallelSize,MaxModelLen,Dataset,InputLen,OutputLen
+v6e-1,Qwen/Qwen2.5-1.5B-Instruct,256,4096,1,2048,sonnet,1800,128
+v6e-1,meta-llama/Llama-3.1-8B-Instruct,256,1024,1,2048,sonnet,1800,128
```
Comment (QiliangCui) [inline cases/hourly_jax.csv:3]: Please make sure one empty line at the end of the file... 
And only one...
There should be 4 lines in this file if there are 2 cases. 

Otherwise, the script. will miss the last test case.

File: scripts/scheduler/hourly_run.sh [hunk 3]
```diff
@@ -16,12 +16,18 @@ else
...
 # Eventually, TorchAx and vLLM should run the same test case.
 # for now, we start from v6e-1.
 echo "./scripts/scheduler/create_job.sh ./cases/hourly_torchax.csv \"\" $TAG HOURLY_TORCHAX TPU_COMMONS"
 ./scripts/scheduler/create_job.sh ./cases/hourly_torchax.csv "" $TAG HOURLY_TORCHAX TPU_COMMONS
 
+# Run TPU Commons + JAX test.
+# Eventually, JAX and vLLM should run the same test case.
+# for now, we start from v6e-1.
+echo "./scripts/scheduler/create_job.sh ./cases/hourly_jax.csv \"\" $TAG HOURLY_TORCHAX TPU_COMMONS jax"
+./scripts/scheduler/create_job.sh ./cases/hourly_jax.csv "" $TAG HOURLY_TORCHAX TPU_COMMONS jax
```
Comment (QiliangCui) [inline scripts/scheduler/hourly_run.sh:29]: HOURLY_TORCHAX  => HOURLY_JAX

we need to diferentiate the Jax and Jax Torchax and Normal VLLM so that we can build comparing like go/vllm-bm-hourly-compare.  

when you submit the "HOURLY_JAX", 2 things to follow(i can do it quickly if you have no interest...)
1. create a graph similar to go/vllm-torchax-bm-hourly but with backend=jax data.
2. add the the jax to go/vllm-bm-hourly-compare

**Review by QiliangCui — CHANGES_REQUESTED** [review]
